Introduce tests to run after upgrade

mirekdlugosz opened the pull request:
These tests focus around the question: if I upgrade Quipucords to newest version, can I still use it?

They modify existing data, reuse it and create brand new data. Data reading is covered by snapshot tests, as creating a new snapshot fundamentally reads the data through API.

This PR includes #560 and will be rebased after the other one is merged. I included #560 to avoid conflicts.

Some of the new tests would end up being run by PR pipeline, which is not what we want. This will be fixed on discovery-ci side in next step.

mirekdlugosz committed:
Introduce tests to run after upgrade

These tests focus around the question: if I upgrade Quipucords to newest
version, can I still use it?

They modify existing data, reuse it and create brand new data. Data
reading is covered by snapshot tests, as creating a new snapshot
fundamentally reads the data through API.

bruno-fs commented on camayoc/tests/qpc/cli/test_credentials.py:
```diff
@@ -710,3 +710,69 @@ def test_clear_all(isolated_filesystem, qpc_server_config, cleaning_data_provide
+    except ValueError:
+        pytest.skip("There are no credentials of this type")
+
+    credential = random.choice(all_credentials)
+    credential_name = credential.get("name")
+    if not credential.get("username"):
+        pytest.skip("Credential does not store username")
```
I'm probably missing context here, but I'm under the impression we might want to only skip if no credentials have `username`. I suggest we randomly select from credentials pre-filtered, with the filter being "credential has `username` key".

mirekdlugosz replied:
You are right. Fixed in 4944aea826887cd7b2ae4420083e16bca0a592fa

ruda reviewed (approved):
LGTM.

mirekdlugosz committed:
Make tests editing existing credentials/sources more robust

Wrap credential / source name in single quotes to ensure test may work
with objects that have spaces in names.

mirekdlugosz committed:
Make test_rerun_scanjob more robust

Database dump may have scans with multiple sources, including sources of
different types. When re-running Ansible scan, we might actually re-run
a scan that has both Ansible and network source, in which case it will
have ansible logs in report archive.

Instead of relying on source_type fixture value, let's look into actual
scan source types.
